disco_aws_automation/disco_vpc_peerings.py

```diff
@@ -26,81 +29,95 @@ class DiscoVPCPeerings(object):
+            self._create_peering_connections(desired_peerings - existing_peerings)
+            self._create_peering_routes(desired_peerings)
 
-    def _get_existing_peerings(self):
+    def _get_existing_peerings(self, vpc):
+        """
+        Get the set of PeeringConnections for the existing peerings for given DiscoVPC object
+        """
         current_peerings = set()
+        for peering in self.list_peerings(vpc.get_vpc_id()):
 
-        for peering in DiscoVPCPeerings.list_peerings(self.disco_vpc.get_vpc_id()):
-
-            peer_vpc_id = self._get_peer_vpc_id(peering)
-            peer_vpc = self._find_peer_vpc(peer_vpc_id)
+            peer_vpc = self._find_peer_vpc(self._get_peer_vpc_id(vpc.get_vpc_id(), peering))
             if not peer_vpc:
                 logger.warning("Failed to find the peer VPC (%s) associated with peering (%s). "
                                "If the VPC no longer exists, please delete the peering manually.",
-                               peer_vpc_id, peering['VpcPeeringConnectionId'])
+                               peer_vpc['VpcId'], peering['VpcPeeringConnectionId'])
                 continue
 
-            peering_query = create_filters(
-                {'route.vpc-peering-connection-id': [peering['VpcPeeringConnectionId']]}
-            )
-
-            vpc_peering_route_tables = throttled_call(self.boto3_ec2.describe_route_tables,
-                                                      Filters=peering_query)['RouteTables']
+            vpc_peering_route_tables = throttled_call(
+                self.client.describe_route_tables,
+                Filters=create_filters({
+                    'route.vpc-peering-connection-id': [peering['VpcPeeringConnectionId']]
+                })
+            )['RouteTables']
 
             for route_table in vpc_peering_route_tables:
                 tags_dict = tag2dict(route_table['Tags'])
 
                 subnet_name_parts = tags_dict['Name'].split('_')
-                if subnet_name_parts[0] == self.disco_vpc.environment_name:
-                    source_network = subnet_name_parts[0] + ':' + \
-                        self.disco_vpc.environment_type + '/' + \
-                        subnet_name_parts[1]
-
-                    route_cidrs = [route['DestinationCidrBlock']
-                                   for route in route_table['Routes']
-                                   if route.get('VpcPeeringConnectionId') ==
-                                   peering['VpcPeeringConnectionId']]
-
-                    for network in peer_vpc.networks.values():
-                        if str(network.network_cidr) in route_cidrs:
-                            dest_network = peer_vpc.environment_name + ':' + \
-                                peer_vpc.environment_type + '/' + network.name
-
-                            current_peerings.add(source_network + ' ' + dest_network)
+                if subnet_name_parts[0] == vpc.environment_name:
```

Comment:
I'd splat out subset_name_parts into variables so it's easier to understand what it's composed of.